packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -302,3 +313,12 @@ def parse_args(self, message):
             self.reply(response, message)
             return None
         return args
+
+    @property
+    def _output_dir(self):
```

Comment:
1. I don't see any reason this should be private; can this be renamed to `output_dir` (without the leading underscore)?
2. Can you add a type hint for the return type? See below.

```suggestion
    def output_dir(self) -> str:
```